utils/loading_cache.hh

```diff
@@ -290,6 +290,34 @@ public:
         });
     }
 
+    value_ptr get_ptr(const Key& k, value_type value) {
```

Comment:
The name is strange. You have a value, why do you call get? And why do you return a value_ptr?

My expectation was for void put(key, value) (symmetric with get(key) -> value).

Maybe you have a good reason for it, but please explain it, and add comments explaining the function's use.

Reply:
I'll change it, `put` seems better for describing what actually happens